pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/HttpSegmentFetcher.java

```diff
@@ -90,6 +91,54 @@ public void fetchSegmentToLocal(URI downloadURI, File dest)
     });
   }
 
+  @Override
+  public File fetchUntarSegmentToLocalStreamed(URI downloadURI, File dest, long rateLimit)
```

Comment:
I think return type can be void since you are providing the pointer to dest ?

Reply:
Discussed offline. We need to return the pointer to the untarred dir